package/MDAnalysis/exceptions.py

```diff
@@ -35,6 +35,26 @@ class SelectionError(Exception):
 class NoDataError(ValueError, AttributeError):
     """Raised when empty input is not allowed or required data are missing.
 
+    This exception is used in various scenarios:
+    -   Raised when a Topology Attribute (e.g., bonds, charges) is not
+        present in the data.
+
+    -   Raised when data is missing in an analysis class because the `run()`
+        method has not been called. Examples include polymer analysis, PSA
+        (Potential Surface Analysis), dielectric analysis, and hydrogen bond
+        analysis.
+
+    -   Raised when timestep data is missing, such as positions, velocities,
+        or forces.
+
+    -   Raised in the `nojump` transformation if there is no box information
+        in the universe.
+
+    Additionally, this exception may be raised in cases where:
```

Comment:
If my original comment is still accurate (I haven't checked, so please correct me if I'm wrong) -- it says that this exception is *not* raised in these cases. In which case I would recommend rephrasing to the below (and keeping it as a sentence to save some vertical space):

```suggestion
    This exception should not be raised in cases where arrays have zero width, or AtomGroups are empty.
```

Reply:
@lilyminium I found that AtomGroups generate NoDataError [here](https://github.com/MDAnalysis/mdanalysis/blob/37d8fee8e54af08aafe1ad1349811a714bbd87e2/benchmarks/benchmarks/ag_methods.py#L246), but not able to locate for Zero width array, so what should i document ?

Reply:
@HeetVekariya that function does not raise `NoDataError` but simply ignores the error if it is triggered some other way. To see if code raises `NoDataError`, you'd want to search for `raise NoDataError`. If you can't find any other instances of it being raised, I would say go with what I've suggested here and document that `NoDataError` is *not* raised in these circumstances.